project1/server/server_tasks/command_ids.h

```diff
@@ -0,0 +1,24 @@
+/**
+ * @File The thread safe data structure to store active command ID's
+ */
+
+#ifndef PROJECT1_COMMAND_IDS_H
+#define PROJECT1_COMMAND_IDS_H
+
+#include <mutex>
+#include <algorithm>
+#include <list>
+namespace server_tasks {
+    class CommandIDs {
+    public:
+
+        void Insert(uint16_t id);
+        bool Contains(uint16_t id);
+        void Delete(uint16_t id);
+
+    private:
+        std::list<uint16_t> command_ids_;
```

Comment:
How about using an `std::unordered_set` here internally? It's more efficient, and will be easier for people reading your code to interpret.